model/api_operations.js

```diff
@@ -66,6 +66,58 @@ function get_user_info(name)
     });
 }
 
+function submit_video(video_id, which_channel, seconds, who_submit) {
+    // create a new entry in the table named "video"
+    // also create a new entry in the table named "voting"
+    // using findOrCreate https://sequelize.readthedocs.io/en/2.0/docs/models-usage/
+    /*User
+        .findOrCreate({ where: { username: 'sdepold' }, defaults: { job: 'Technical Lead JavaScript' } })
+        .spread((user, created) => {
+            console.log(user.get({
+                plain: true
+            }))
+            console.log(created)
+
+            /*
+             findOrCreate returns an array containing the object that was found or created and a boolean that will be true if a new object was created and false if not, like so:
+        
+            [ {
+                username: 'sdepold',
+                job: 'Technical Lead JavaScript',
+                id: 1,
+                createdAt: Fri Mar 22 2013 21: 28: 34 GMT + 0100(CET),
+                updatedAt: Fri Mar 22 2013 21: 28: 34 GMT + 0100(CET)
+              },
+              true ]
+        
+         In the example above, the "spread" on line 75 divides the array into its 2 parts and passes them as arguments to the callback function defined beginning at line 39, which treats them as "user" and "created" in this case. (So "user" will be the object from index 0 of the returned array and "created" will equal "true".)
+            *//*
+        }) */
+    return model.video.findOrCreate({ 
+        where: { id: video_id, channel: which_channel }, // where same video appear in the same channel twice
+        defaults: { by: who_submit, length: seconds } // if the video does not exist yet, we will create it with person = user
+    }).spread((vid, created) => {
+        console.log(vid.get({
+            plain: true
+        }))
+        console.log(created)
+        return [vid, created];
+    })
+}
+function get_video_list(channel_id) {
+    // return a list of video submitted in the channel
+    return model.video.findAll({
+        where: {channel: channel_id},
+        raw: true
+    }).then((res) => 
+    {
+        return res;
+    }).catch((err) =>
+    {
+        throw err;
+    });
+}
+
 function create_channel(user_id)
 {
     return model.channel.destroy({ where : { host : user_id }})
```

Comment:
If a channel host user submits a video to be saved in `video` and later attempts to create a channel again, it will produce channel creation error since it will not be able to `destroy` or delete their previous channel since the channel id is a foreign key in the video model. It will produce error like so: `error: update or delete on table "channels" violates foreign key constraint "videos_channel_fkey" on table "videos"`

So channel creation (and previous channel deletion) needs all `video` model data deletion for that channel as well.

Reply:
Yes, all videos associated with that channel. The videos will have to be deleted before channel deletion.

Reply:
The other option would be to not make channel a foreign key in `video` but then we will have to ensure integrity of the data some other way.

Reply:
This will happen when the person is the host. When a normal user create a channel, the video that he submitted don't have to be destroy. So I will add a condition to check if that user is a host, if the user is a host, we simply destroy all the video belong to the channel.